File: Maverick/Level1/MBTI_Test.py
def solution(survey, choices):
    answer = ''
    mbti = {'R':0,'T':0,'C':0,'F':0,'J':0,'M':0,'A':0,'N':0}
    for i in range(len(survey)):
        l = survey[i][0]
        r = survey[i][1]
        if choices[i]-4>0:
            mbti[r] += choices[i]-4
        elif choices[i] - 4 <0:
            mbti[l] += 4 - choices[i]
    answer += 'R' if mbti['R']>=mbti['T'] else 'T'
    answer += 'C' if mbti['C']>=mbti['F'] else 'F'
    answer += 'J' if mbti['J']>=mbti['M'] else 'M'
    answer += 'A' if mbti['A']>=mbti['N'] else 'N'

    return answer

survey = list(map(str,input().split()))
choices = list(map(int,input().split()))

print(survey)
print(choices)
print(solution(survey,choices))

File: Cindy/Level1/MBTI_Test.js
function solution(survey, choices) {
  const types = { R: 0, T: 0, C: 0, F: 0, J: 0, M: 0, A: 0, N: 0 };
  let answer = "";
  for (let i = 0; i < choices.length; i++) {
    const [leftType, rightType] = survey[i].split("");
    const result = Math.abs(choices[i] - 4);
    if (choices[i] < 4) {
      types[leftType] += result;
    } else if (choices[i] > 4) {
      types[rightType] += result;
    }
  }
  const type = Object.keys(types);
  for (let i = 0; i < type.length; i += 2) {
    const leftType = types[type[i]];
    const rightType = types[type[i + 1]];
    if (leftType >= rightType) {
      answer += type[i];
    } else {
      answer += type[i + 1];
    }
  }
  return answer;
}
